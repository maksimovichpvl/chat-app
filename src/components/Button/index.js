import React from 'react';
import PropTypes from 'prop-types';

import './styles.css';

const Button = ({ title, onClick }) => (
  <button 
    type="button"
    onClick={(e) => onClick()}
    className="btn btn-primary"
  >
    {title}
  </button>
);

Button.propTypes = {
  title: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default Button;