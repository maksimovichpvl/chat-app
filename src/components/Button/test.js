import React from 'react';
import renderer from 'react-test-renderer';

import Button from './index.js';
import { mount } from 'enzyme';

describe('Button Component', function() {
  const onClickActionMock = jest.fn();
  const titleMock = "Test title";

  it('renders without crashing', () => {
    const tree = renderer
      .create(<Button  title={titleMock} onClick={onClickActionMock} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Correctly props', () => {
    const wrapper = mount(<Button  title={titleMock} onClick={onClickActionMock} />);
    expect(wrapper.props().title).toEqual(titleMock);
    expect(wrapper.props().onClick).toEqual(onClickActionMock);
  });
})


