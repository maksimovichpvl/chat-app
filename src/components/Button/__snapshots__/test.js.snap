// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders without crashing 1`] = `
<button
  className="btn btn-primary"
  onClick={[Function]}
  type="button"
>
  Test title
</button>
`;
